components/brave_wallet/common/eth_request_helper.cc

```diff
@@ -106,6 +107,39 @@ const char kRequestJsonRPC[] = "2.0";
 
 namespace brave_wallet {
 
+namespace {
+// EIP-712 type hash for Order struct used in CoW swap
+//
+// keccak256("Order(address sellToken,address buyToken,address receiver,
+//                  uint256 sellAmount,uint256 buyAmount,uint32 validTo,
+//                  bytes32 appData,uint256 feeAmount,string kind,
+//                  bool partiallyFillable,string sellTokenBalance,
+//                  string buyTokenBalance)")
+constexpr char kCowSwapTypeHash[] =
+    "D5A25BA2E97094AD7D83DC28A6572DA797D6B3E7FC6663BD93EFB789FC17E489";
+
+mojom::EthSignTypedDataMetaPtr ParseCowSwapOrder(
+    const base::Value::Dict& value) {
```

Comment:
blocking question: Can we assume this type here? Below it's being set to auto on line 513.

Reply:
L513 already ensures it:

```cpp
  const auto* message = dict->FindDict("message");
  if (!message) {
    return false;
  }
```

`FindDict()` returns a `nullptr` if the value for the `message` key is not a dictionary. Source: https://chromium.googlesource.com/chromium/src/+/HEAD/base/values.cc#465

Reply:
Ahh, thanks I didn't check that implementation as well for this. I appreciate you pointing that out.